desktop/api/src/main/java/org/datacleaner/actions/PreviewTransformedDataActionListener.java

```diff
@@ -162,6 +167,8 @@ protected PreviewJob createPreviewJob() {
                     ajb.removeSourceTable(otherTable);
                 }
             }
+            alreadyFiltered = sourceColumnFinder.findAllSourceJobs(tjb).stream().filter(
+                    o -> o instanceof HasFilterOutcomes).findAny().isPresent();
```

Comment:
Here, the extra scoping is starting to do more harm than good. It would be quite a bit more readable if alreadyFiltered was declared and used where it is used.